packages/compiler/src/testing/tester.ts

```diff
@@ -60,28 +59,16 @@ async function createTesterFs(base: string, options: TesterOptions) {
     },
   };
 
-  const sl = await createSourceLoader(host);
+  const sl = await createSourceLoader(host, {
+    resolve: {
+      conditions: ["typespec", "import", "default"],
+    },
+  });
   const selfName = JSON.parse(await readFile(resolvePath(base, "package.json"), "utf8")).name;
   for (const lib of options.libraries) {
     await sl.importPath(lib, NoTarget, base);
-
-    const resolved = await resolveModule(
-      {
-        realpath: async (x) => x,
-        stat: NodeHost.stat,
-        readFile: async (path) => {
-          const file = await NodeHost.readFile(path);
-          return file.text;
-        },
-      },
-      lib,
-      { baseDir: base, conditions: ["import", "default"] },
-    );
-    if (resolved.type === "module") {
-      const virtualPath = computeRelativePath(lib, resolved.mainFile);
-      fs.addJsFile(virtualPath, host.getJsImport(resolved.mainFile));
```

Comment:
I think I previously only half the job. This also needs to register the package.json.

I think with the ability to pass custom conditions this makes everything much simpler.